### PR TITLE
🚀 4.2.0 release

### PR DESCRIPTION
**Pull request by OrtnerMichael**
release pr

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ All notable changes to magpylib are documented here.
 
 # Releases
 
-## [4.2.0] - XXXXXXXXXXX
+## [4.2.0] - 2023-01-27
 - Change pyvista plotting defaults when using `show(backend='pyvista')` to fit better with other libraries. ([#551](https://github.com/magpylib/magpylib/issues/551))
 - (Re)introducing the powerful `misc.Triangle` class that can be used to compute magnetic fields of arbitrarily shaped bodies by approximating their surface with triangular facets. ([#568](https://github.com/magpylib/magpylib/issues/568))
```
Comment (Alexboiboi) [inline CHANGELOG.md:8]: I'd put the Triangle PR first , since the pyvista PR is not so much relevant for this release

**Review by Alexboiboi — CHANGES_REQUESTED** [review]
otherwise ok

File: CHANGELOG.md (hunk 1 above)
Comment (Alexboiboi) [reply CHANGELOG.md:8]: @OrtnerMichael also please execute `pre-commit run --all` before

Comment (Alexboiboi) [reply CHANGELOG.md:7]: I also updated some test to reach maximum coverage

**Review by Alexboiboi — APPROVED** [review]
